packages/terra-clinical-onset-picker/src/OnsetPicker.jsx

```diff
@@ -0,0 +1,427 @@
+        return { value: year.toString(), display: year.toString() };
+      },
+    );
+  }
+
+  /**
+   * Calulcates maximum number allowed for each age duration.
+   * Will use the lesser of power of two or the difference between birth for week/month.
+   * Power of two for maximum weeks/months (8 weeks = 2 months, 24 months = 2 years)
+   */
+  maxAgeCount() {
+    switch (this.state.ageCalcDuration) {
+      case 'years':
+        return moment().diff(moment(this.props.birthdate), 'years');
+      case 'months': {
+        const monthDiff = moment().diff(moment(this.props.birthdate), 'months');
+        return monthDiff > 24 ? 24 : monthDiff;
+      }
+      default: {
+        const weekDiff = moment().diff(moment(this.props.birthdate), 'weeks');
+        return weekDiff > 8 ? 8 : weekDiff;
+      }
+    }
+  }
+
+  /*
+   * Create object to pass to SelectField for age duration options.
+   * Does not add durations that are not possible based on birth.
+   */
+  allowedAgeDurations() {
+    const ageMoment = moment(this.props.birthdate);
+
+    if (moment().diff(ageMoment, 'weeks') === 0) {
+      return null;
+    }
+
+    const ageDurations = [{ value: 'weeks', display: 'Week(s)' }];
+
+    // Do not add month option if less than a month old
+    if (moment().diff(ageMoment, 'months') > 0) {
+      ageDurations.push({ value: 'months', display: 'Month(s)' });
+    }
+
+    // Do not add year option if less than a month old
+    if (moment().diff(ageMoment, 'years') > 0) {
+      ageDurations.push({ value: 'years', display: 'Year(s)' });
+    }
+
+    return ageDurations;
+  }
+
+  render() {
+    const {
+      birthdate,
+      granularity,
+      granularitySelectName,
+      granularitySelectOnChange,
+      precision,
+      precisionSelectName,
+      precisionSelectOnChange,
+      onsetDate,
+      onsetDateInputName,
+      onsetDateInputOnChange,
+      ...customProps
+    } = this.props;
+
+    const intl = this.context.intl;
+
+    let granularitySelect = null;
+    if (this.state.precision !== 'UNKNOWN') {
+      granularitySelect = (<SelectField
+        options={[{ value: 'MONTH', display: intl.formatMessage({ id: 'Terra.onsetPicker.granularityMonth' }) },
+                  { value: 'YEAR', display: intl.formatMessage({ id: 'Terra.onsetPicker.granularityYear' }) },
+                  { value: 'AGE', display: intl.formatMessage({ id: 'Terra.onsetPicker.granularityAge' }) },
+                  { value: 'DATE', display: intl.formatMessage({ id: 'Terra.onsetPicker.granularityDate' }) }]}
+        name={this.props.granularitySelectName}
+        defaultValue={this.state.granularity}
+        onChange={this.changeGranularity}
+        isInline
+      />);
+    }
+
+    let ageCalcCountSelect = null;
+    let ageCalcDurationSelect = null;
+    if (this.state.granularity === 'AGE' && this.state.precision !== 'UNKNOWN') {
+      ageCalcCountSelect = (<NumberField
+        className="terra-OnsetPicker-ageCalcCount"
```

Comment:
I don't see any scss files. Are these classes being used at all?

Reply:
If there aren't any styles specific to this component then remove the classes otherwise, please convert to use css modules and updated naming convention I mentioned in tech design: https://github.com/cerner/terra-clinical/pull/184

Reply:
If there are no styling with this component, but naming is desired for structural consistency, use `data-terra-clinical-onset-picker-'name_of_piece'`

Reply:
Since these had no stylistic purpose and were used for structural consistency, I have changed them to data attributes as per @emilyrohrbough suggestion.